File: app/src/main/java/com/sk/revisit/managers/MySettingsManager.java
package com.sk.revisit.managers;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.preference.PreferenceManager;

public class MySettingsManager {
	private static final String
			KEY_ROOT_PATH = "rootPath",
			KEY_THEME = "theme",
			KEY_DN_PATH = "dnPath",
			KEY_IS_FIRST = "isFirst",
			KEY_USR_AGENT_STR = "userAgentCustom",
			KEY_USR_AGENT_DEF = "userAgentDefaults",
			KEY_MAX_WEB_TIMEOUT = "webTimeoutDuration";

	private final SharedPreferences prefs;

	public MySettingsManager(Context context) {
		prefs = PreferenceManager.getDefaultSharedPreferences(context);
	}

	void setTheme(String val) {
		putStr(KEY_THEME, val);
	}

	void setUserAgentCustom(String val) {
		putStr(KEY_USR_AGENT_STR, val);
	}

	public String getRootStoragePath() {
		return prefs.getString(KEY_ROOT_PATH, null);
	}

	public void setRootStoragePath(String folderPath) {
		putStr(KEY_ROOT_PATH, folderPath);
	}

	public String getDownloadStoragePath() {
		return prefs.getString(KEY_DN_PATH, null);
	}

	public boolean getIsFirst() {
		return prefs.getBoolean(KEY_IS_FIRST, true);
	}

	public void setIsFirst(boolean o) {
		putBoolean(KEY_IS_FIRST, o);
	}

	void putStr(String key, String val) {
		prefs.edit().putString(key, val).apply();
	}

	void putInt(String key, int val) {
		prefs.edit().putInt(key, val).apply();
	}

	void putBoolean(String key, boolean val) {
		prefs.edit().putBoolean(key, val).apply();
	}
}

File: app/src/main/java/com/sk/revisit/managers/MySettingsManager.kt
package com.sk.revisit.managers

import android.content.Context
import android.content.SharedPreferences
import androidx.preference.PreferenceManager

enum class KEY {
    ROOT_PATH,
    THEME,
    DN_PATH,
    IS_FIRST,
    USR_AGENT_STR,
    USR_AGENT_DEF,
    MAX_WEB_TIMEOUT,
}

open class MySettingsManager(context: Context) : SettingsHelp(PreferenceManager.getDefaultSharedPreferences(context)) {

    var rootStoragePath: String
        get() = getStr(KEY.ROOT_PATH, "/sdcard/Android/obb/com.sk.revisit")!!
        set(folderPath) = putStr(KEY.ROOT_PATH, folderPath)

    val downloadStoragePath: String?
        get() = getStr(KEY.DN_PATH, null)

    var isFirst: Boolean
        get() = getBoolean(KEY.IS_FIRST, true)
        set(valValue) = putBoolean(KEY.IS_FIRST, valValue)

    fun setTheme(valValue: String) {
        putStr(KEY.THEME, valValue)
    }

    fun setUserAgentCustom(valValue: String) {
        putStr(KEY.USR_AGENT_STR, valValue)
    }
}

open class SettingsHelp(protected val prefs: SharedPreferences) {

    fun getStr(key: KEY, def: String?): String? {
        return prefs.getString(key.name, def)
    }

    fun getBoolean(key: KEY, def: Boolean): Boolean {
        return prefs.getBoolean(key.name, def)
    }

    fun putStr(key: KEY, valValue: String) {
        prefs.edit().putString(key.name, valValue).apply()
    }

    fun putInt(key: KEY, valValue: Int) {
        prefs.edit().putInt(key.name, valValue).apply()
    }

    fun putBoolean(key: KEY, valValue: Boolean) {
        prefs.edit().putBoolean(key.name, valValue).apply()
    }
}
